File: src/types/appEntry.ts
export interface AppEntry {
    id: string;               // Unique identifier (UUID)
    name: string;             // Display name
    bundleId: string;         // macOS bundle identifier
    path: string;             // Full app path
    iconPath?: string;        // Optional icon path
    position: number;         // Position index (1-based)
    disabled: boolean;        // Whether shortcut is disabled
  }

File: src-tauri/src/app_entry.rs
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct AppEntry {
    pub id: String,
    pub name: String,
    pub bundle_id: String,
    pub path: String,
    pub icon_path: Option<String>,
    pub position: u32,
    pub disabled: bool,
}